hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/ServerRpcConnection.java

```diff
@@ -688,41 +691,75 @@ private void doBadPreambleHandling(String msg) throws IOException {
   }
 
   private void doBadPreambleHandling(String msg, Exception e) throws IOException {
-    RpcServer.LOG.warn(msg);
+    RpcServer.LOG.warn(msg, e);
     doRespond(getErrorResponse(msg, e));
   }
 
+  private boolean doConnectionRegistryResponse() throws IOException {
+    if (!(rpcServer.server instanceof ConnectionRegistryEndpoint)) {
+      // should be in tests or some scenarios where we should not reach here
+      return false;
+    }
+    // on backup masters, this request may be blocked since we need to fetch it from filesystem,
+    // but since it is just backup master, it is not a critical problem
+    String clusterId = ((ConnectionRegistryEndpoint) rpcServer.server).getClusterId();
```

Comment:
is there no way to protect this from blocking? maybe in a followup since this PR is already large. 

my concern is that backup masters may soon become active, we don't want them blocked.

Reply:
The request is blocked, not the backup masters are blocked.